### PR TITLE
Ghcjs 8.2 & 8.4

### PR DESCRIPTION
**Pull request by ElvishJerricco**
###### Motivation for this change

Adds GHCJS 8.2 and GHCJS 8.4.

These builds are currently blocked on ghcjs/ghcjs#654. It's important that these builds be checked more than once until that bug is fixed, as it is an intermittent failure.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ElvishJerricco** [comment]
Seems rebasing caused the build to fail.

**Comment by ElvishJerricco** [comment]
Ah, no I just had a dirty work tree. Builds fine!

**Comment by ElvishJerricco** [comment]
Adding GHCJS 8.4 as well, since they're closely related and both have build failures

**Comment by ElvishJerricco** [comment]
Ok we have what seems to be a working workaround for the panic. So I think this is good to merge. @peti what do you think?

File: pkgs/development/compilers/ghcjs-ng/mk-stage0.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+, runCommand
+, cabal2nix
+, yq
+}:
+
+runCommand "stage0.nix" {
+  buildInputs = [cabal2nix yq];
+} ''
+  (
+    printf '{ callPackage, configuredSrc }:\n\n{\n\n'
```
Comment (3noch) [inline pkgs/development/compilers/ghcjs-ng/mk-stage0.nix]: Might be worth a comment explaining what this is doing O_O

**Review by peti — APPROVED** [review]

**Comment by ElvishJerricco** [comment]
@peti are you sure? The expressions are on the nixos-unstable channel but they are not cached.

**Comment by peti** [comment]
They haven't compiled on Hydra successfully yet: https://hydra.nixos.org/eval/1458249?filter=ghcjs&compare=1457965

**Comment by peti** [comment]
@ElvishJerricco, neither of these builds has ever succeeded on Hydra:

 - https://hydra.nixos.org/job/nixpkgs/haskell-updates/haskell.compiler.ghcjs84.x86_64-linux
 - https://hydra.nixos.org/job/nixpkgs/haskell-updates/haskell.compiler.ghcjs82.x86_64-linux

Can you please take a look at these issues? If the  package does not build, then we should have this code in Nixpkgs and I'm inclined to revert these changes if that state does not change.

**Comment by ElvishJerricco** [comment]
Right, sorry. Had to wait on the GHC 8.0 fix. Though that build error is super obscure. The log is a binary??

**Comment by 3noch** [comment]
This is now in nixpkgs-unstable.

**Pull request merged**